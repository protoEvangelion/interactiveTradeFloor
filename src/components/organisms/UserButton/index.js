import { Button } from 'components'
import { LoginModal } from 'containers'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

const InnerButton = styled.div`
  display: flex;
  align-items: center;
`

const Image = styled.img`
  margin-right: 0.5rem;
  border-radius: 50%;
`

const UserButton = ({ authenticated, user, onLogin, onLogout, ...props }) => {
  return (
    <div>
      {authenticated &&
        <Button {...props} onClick={onLogout}>
          <InnerButton>
            <Image src={user.picture} width={25} height={25} />
            Sign out
          </InnerButton>
        </Button>
      }
      {/* istanbul ignore next */ !authenticated && <Button {...props} onClick={onLogin}>Sign in</Button>}
      <LoginModal />
    </div>
  )
}

UserButton.propTypes = {
  authenticated: PropTypes.bool.isRequired,
  user: PropTypes.shape({
    picture: PropTypes.string.isRequired,
  }),
  onLogin: PropTypes.func.isRequired,
  onLogout: PropTypes.func.isRequired,
}

export default UserButton
