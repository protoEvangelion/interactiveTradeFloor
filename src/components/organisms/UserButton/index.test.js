import React from 'react'
import renderer from 'react-test-renderer'
import UserButton from '.'

const props = {
  authenticated: false,
  user: { picture: 'test.jpg' },
  onLogin: jest.fn(),
  onLogout: jest.fn(),
}

it('renders User button snapshot', () => {
  const tree = renderer.create(
    <UserButton {...props} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
