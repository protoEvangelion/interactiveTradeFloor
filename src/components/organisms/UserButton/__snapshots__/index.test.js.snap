// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders User button snapshot 1`] = `
<div>
  <span>
    <div
      className="dqUvXD"
    >
      <img
        className="hOOyAl"
        height={25}
        src="test.jpg"
        width={25}
      />
      Sign out
    </div>
  </span>
  <span />
</div>
`;
